libs/agno/agno/vectordb/milvus/milvus.py

```diff
@@ -184,61 +334,68 @@ def id_exists(self, id: str) -> bool:
-                "meta_data": document.meta_data,
-                "content": cleaned_content,
-                "usage": document.usage,
-            }
-            self.client.insert(
-                collection_name=self.collection,
-                data=data,
-            )
-            log_debug(f"Inserted document: {document.name} ({document.meta_data})")
+        if self.search_type == SearchType.hybrid:
+            for document in documents:
+                self._insert_hybrid_document(document)
+        else:
+            for document in documents:
+                document.embed(embedder=self.embedder)
+                data = self._prepare_document_data(document, include_vectors=True)
+
+                self.client.insert(
+                    collection_name=self.collection,
+                    data=data,
+                )
+                log_debug(f"Inserted document: {document.name} ({document.meta_data})")
+
+        log_info(f"Inserted {len(documents)} documents")
 
     async def async_insert(self, documents: List[Document], filters: Optional[Dict[str, Any]] = None) -> None:
-        """Insert documents asynchronously with controlled concurrency."""
-        log_debug(f"Inserting {len(documents)} documents asynchronously")
+        """Insert documents asynchronously based on search type."""
+        log_info(f"Inserting {len(documents)} documents asynchronously")
 
-        async def process_document(document):
-            # Same processing code as before
-            document.embed(embedder=self.embedder)
-            cleaned_content = document.content.replace("\x00", "\ufffd")
-            doc_id = md5(cleaned_content.encode()).hexdigest()
-            data = {
-                "id": doc_id,
-                "vector": document.embedding,
-                "name": document.name,
-                "meta_data": document.meta_data,
-                "content": cleaned_content,
-                "usage": document.usage,
-            }
-            await self.async_client.insert(
-                collection_name=self.collection,
-                data=data,
-            )
-            log_debug(f"Inserted document asynchronously: {document.name} ({document.meta_data})")
-            return data
+        if self.search_type == SearchType.hybrid:
+            await asyncio.gather(*[self._async_insert_hybrid_document(doc) for doc in documents])
+        else:
 
-        await asyncio.gather(*[process_document(doc) for doc in documents])
+            async def process_document(document: Document) -> Dict[str, Any]:
+                document.embed(embedder=self.embedder)
+                data = self._prepare_document_data(document, include_vectors=True)
+
+                await self.async_client.insert(
+                    collection_name=self.collection,
+                    data=data,
+                )
+                log_debug(f"Inserted document asynchronously: {document.name} ({document.meta_data})")
```

Comment:
This also can be a call to `_async_insert_hybrid_document `

Reply:
Addressed, same as above comment.